sycl/test-e2e/Plugin/cuda_queue_priority.cpp

```diff
@@ -0,0 +1,47 @@
+// REQUIRES: gpu, cuda
+
+// RUN: %{build} %cuda_options -o %t.out
+// RUN: %{run} %t.out
+//
+// Check that queue priority is passed to CUDA runtime
+#define SYCL_EXT_ONEAPI_BACKEND_CUDA_EXPERIMENTAL 1
+#include <sycl/ext/oneapi/experimental/backend/cuda.hpp>
+#include <sycl/sycl.hpp>
+
+#include <cuda.h>
+
+#include <assert.h>
+
+int get_real_priority(sycl::context &C, sycl::device &D,
+                      sycl::property_list Props) {
+  sycl::queue Q(C, D, Props);
+  CUstream QNative = sycl::get_native<sycl::backend::ext_oneapi_cuda>(Q);
```

Comment:
Note that this only returns one of the streams of the queue.

Is this extension meant to be used with the `in_order` property as well? So the SYCL queues map to a single CUDA stream?

From this patch it looks like it would work with out-of-order queues, but that seems a bit less intuitive.

Reply:
> Is this extension meant to be used with the in_order property as well? So the SYCL queues map to a single CUDA stream?

No, this extension is not limited to in-order streams. 

> Note that this only returns one of the streams of the queue.

True.

We can call `sycl::get_native` several times in a loop, and then new streams will be created and returned. But I don't think that would add much value to the test.